### PR TITLE
upload cpu mkldnn linux benchmark scripts

### PR DESCRIPTION
**Pull request by OliverLPH**
1. add cpu mkldnn linux benchmark scripts
2. Test scripts contain analysis cpu performance, mkldnn performance, different `cpu_math_library_num_threads` performance. 

reproduce method is as follow 
```shell
# 1. download inference-lib or compile from source codes

# 2. download test codes
git clone https://github.com/PaddlePaddle/continuous_integration.git

# 3. compile test codes
cd continuous_integration/inference/inference_benchmark/cc/Paddle
bash compile.sh all ON OFF OFF /path/of/inference-lib

# 4. prepare python third-party lib
python3.7 -m pip install py3nvml;  # monitor gpu
python3.7 -m pip install cup;          # monitor cpu
python3.7 -m pip install pandas;    # process data
python3.7 -m pip install openpyxl; # process data to excel

# 5. start benchmark tests
bash bin/run_models_benchmark.sh "static" "cpu"

# 6. convert data to excel
python3.7 bin/py_parse_log.py --log_path=./log --output_name=benchmark_mkl_excel.xlsx
```

### CODE REVIEW

**Comment by OliverLPH** [comment]
@lidanqing-intel 
hi, danqing
please help review this PR.
It contains shell scripts `run_models_benchmark.sh`, `run_det_mkl_benchmark.sh` and `run_clas_mkl_benchmark.sh`.

`run_models_benchmark.sh` is main scripts

`run_clas_mkl_benchmark.sh` contains clas/ocr/seg models
```shell
AlexNet      DPN68           GoogLeNet    MobileNetV1             RegNetX_4GF             ResNet50               ShuffleNetV2   Xception41
DarkNet53    EfficientNetB0  HRNet_W18_C  MobileNetV2             Res2Net50_26w_4s        ResNet50_vd            SqueezeNet1_0
DenseNet121  GhostNet_x1_3   InceptionV4  MobileNetV3_large_x1_0  ResNeSt50_fast_1s1x64d  SE_ResNeXt50_vd_32x4d  VGG11
```

```shell
deeplabv3p  fastscnn  hrnet  icnet  pspnet  unet
```

```shell
ch_ppocr_mobile_v1.1_cls_infer  ch_ppocr_mobile_v1.1_det_infer  ch_ppocr_mobile_v1.1_rec_infer
```

`run_det_mkl_benchmark.sh` contains detection models, it runs very slow, so I disable it from `run_models_benchmark.sh`
```shell
faster_rcnn_dcn_r50_vd_fpn_3x_server_side  ppyolo_mobilenet_v3_large   ssd_mobilenet_v1_voc  yolov3_r50vd_dcn
faster_rcnn_r50_1x        yolov3_darknet        yolov4_cspdarknet
mask_rcnn_r50_1x                           ssdlite_mobilenet_v3_large  yolov3_mobilenet_v3
```


Besides these scripts, windows bat scripts would be added to this repo in a few days.

**Review by lidanqing-vv — APPROVED** [review]
LGTM! Thanks @OliverLPH  I have verified reproduction, everything is fine. Thank you very much for providing the scripts

File: inference/inference_benchmark/cc/Paddle/bin/run_models_benchmark.sh [hunk 1]
```diff
@@ -11,12 +11,18 @@ export CASE_ROOT=$ROOT/bin
...
 if [ $# -ge 1 ]; then
     model_type=$1
 fi
 export MODEL_TYPE=${model_type}
 
+# test run-time device
+device_type="gpu"
+if [ $# -ge 2 ]; then
+    device_type=$2
+fi
```
Comment (lidanqing-vv) [inline inference/inference_benchmark/cc/Paddle/bin/run_models_benchmark.sh:25]:
```
wget --no-proxy -q https://sys-p0.bj.bcebos.com/Paddle-UnitTest-Model/PaddleClas.tgz --no-check-certificate	    
tar -zxf PaddleClas.tgz	
```
The downloaded `PaddleClas.tgz` does not contain `AlexNet` and `yolov3_darknet`, it only contains `GhostNet_x1_3`  `Res2Net50_26w_4s`. Is it enough for us to test existing two models ?

Comment (OliverLPH) [reply inference/inference_benchmark/cc/Paddle/bin/run_models_benchmark.sh:25]: Did you fully download `PaddleClas.tgz`? It should contain directory tree as follow. 
```shell
.
├── infer_dygraph
│   ├── AlexNet
│   ├── DarkNet53
│   ├── DenseNet121
│   ├── DPN68
│   ├── EfficientNetB0
│   ├── GhostNet_x1_3
│   ├── GoogLeNet
│   ├── HRNet_W18_C
│   ├── InceptionV4
│   ├── MobileNetV1
│   ├── MobileNetV2
│   ├── MobileNetV3_large_x1_0
│   ├── RegNetX_4GF
│   ├── Res2Net50_26w_4s
│   ├── ResNeSt50_fast_1s1x64d
│   ├── ResNet50
│   ├── ResNet50_vd
│   ├── SE_ResNeXt50_vd_32x4d
│   ├── ShuffleNetV2_x1_0
│   ├── SqueezeNet1_0
│   ├── VGG11
│   ├── VGG16
│   └── Xception41
└── infer_static
    ├── AlexNet
    ├── DarkNet53
    ├── DenseNet121
    ├── DPN68
    ├── EfficientNetB0
    ├── GhostNet_x1_3
    ├── GoogLeNet
    ├── HRNet_W18_C
    ├── InceptionV4
    ├── MobileNetV1
    ├── MobileNetV2
    ├── MobileNetV3_large_x1_0
    ├── RegNetX_4GF
    ├── Res2Net50_26w_4s
    ├── ResNeSt50_fast_1s1x64d
    ├── ResNet50
    ├── ResNet50_vd
    ├── SE_ResNeXt50_vd_32x4d
    ├── ShuffleNetV2
    ├── SqueezeNet1_0
    ├── VGG11
    └── Xception41
```

If you test models which params size less than 100 MB would be fine
```
234M	AlexNet
160M	DarkNet53 # no need to test
32M	DenseNet121
49M	DPN68
21M	EfficientNetB0
29M	GhostNet_x1_3
27M	GoogLeNet
85M	HRNet_W18_C
165M	InceptionV4 # no need to test
17M	MobileNetV1
14M	MobileNetV2
22M	MobileNetV3_large_x1_0
86M	RegNetX_4GF
99M	Res2Net50_26w_4s
101M	ResNeSt50_fast_1s1x64d # no need to test
99M	ResNet50
99M	ResNet50_vd
107M	SE_ResNeXt50_vd_32x4d # no need to test
9.3M	ShuffleNetV2
5.0M	SqueezeNet1_0
507M	VGG11   # no need to test
89M	Xception41
```

Comment (lidanqing-vv) [reply inference/inference_benchmark/cc/Paddle/bin/run_models_benchmark.sh:25]: Hi, @OliverLPH  Thank you, I donwloaded again, and the models are exactly as listed, here. 
now LGTM. Thanks
Running log
```
 ==== start benchmark ====
 AlexNet input image_shape = 3,224,224

start  AlexNet, use_gpu: false, batch_size: 1
finish  AlexNet, use_gpu: false, batch_size: 1


start  AlexNet, use_gpu: false, batch_size: 2
finish  AlexNet, use_gpu: false, batch_size: 2


start  AlexNet, use_gpu: false, batch_size: 4
finish  AlexNet, use_gpu: false, batch_size: 4

 AlexNet input image_shape = 3,224,224

start  AlexNet, use_mkldnn: true, cpu_math_library_num_threads: 1, batch_size: 1
finish  AlexNet, use_mkldnn: true, cpu_math_library_num_threads: 1, batch_size: 1

... ...
```

Comment (OliverLPH) [reply inference/inference_benchmark/cc/Paddle/bin/run_models_benchmark.sh:25]: Brilliant! 
thanks, I will merge this PR

**Pull request merged**